### PR TITLE
Collapse output in function logs

### LINKED ISSUE #1000: Can't collapse output in the function logs after it's expanded

### PR DESCRIPTION
**Pull request by pradel**
Also fixed a prop-types warning in ResultViewer (PropTypes.function is not a valid type) :)
Close #1000 

capture d ecran 2017-06-02 a 16 03 18

### CODE REVIEW

File: src/views/FunctionsView/FunctionLogs/Log.tsx [hunk 1]
```diff
@@ -25,18 +25,29 @@ export default class LogComponent extends React.Component<Props, State> {
...
+            padding-left: 0;
           }
           .graphiql-container :global(div.CodeMirror-lines) {
             @p: .pa0;
           }
         `}</style>
-        <ResultViewer
-          value={message}
-        />
+        <div className='flex itemsBaseline'>
```
Comment (timsuchanek) [inline src/views/FunctionsView/FunctionLogs/Log.tsx]: nice, I never used `itemsBaseline` before!

File: src/views/FunctionsView/FunctionLogs/ResultViewer.js [hunk 2]
```diff
@@ -91,6 +92,11 @@ export class ResultViewer extends React.Component {
       }
     }
   }
+  _onClick = (e) => {
```
Comment (timsuchanek) [inline src/views/FunctionsView/FunctionLogs/ResultViewer.js]: just one note: here in js context the `_` is totally fine, but in typescript we usually just use the `private` keyword

**Comment by timsuchanek** [comment]
Awesome! This looks very good. Thanks for the good work :)

File: src/views/FunctionsView/FunctionLogs/ResultViewer.js (hunk 2 above)
Comment (pradel) [inline src/views/FunctionsView/FunctionLogs/ResultViewer.js]: Okay thanks!

**Pull request merged**